### PR TITLE
[5.1] Merge update parameter and improve contrain checker

### PR DESCRIPTION
**Pull request by bembelimen**
Pull Request for Issue #43100 .

### Summary of Changes
With TUF testing channel is not needed anymore and should use the TUF behaviour. Also allow now explicit testing for "default" and "next"

### Testing Instructions
- Install [Beta 1](https://github.com/joomla/joomla-cms/releases/tag/5.1.0-beta1)
- Apply patch to beta 1
- Go to joomla updater
- Go to options and select "default" + "beta" for stability level
- Update

### Actual result BEFORE applying this Pull Request
Options not there, update does not work

### Expected result AFTER applying this Pull Request
Update works

### CODE REVIEW

**Comment by brianteeman** [comment]
Should the code comments here be updated to include "testing"

https://github.com/joomla/joomla-cms/blob/ea1b3fb967e28369d03fa1e6d1461144533adb8a/administrator/components/com_joomlaupdate/src/Model/UpdateModel.php#L107

File: libraries/src/Updater/ConstraintChecker.php [hunk 1]
```diff
@@ -216,15 +216,11 @@ protected function checkSupportedDatabases(array $supportedDatabases)
      */
     protected function checkStability(string $stability)
     {
-        $minimumStability = ComponentHelper::getParams('com_installer')->get('minimum_stability', Updater::STABILITY_STABLE);
+        $minimumStability = ComponentHelper::getParams('com_joomlaupdate')->get('minimum_stability', Updater::STABILITY_STABLE);
```
Comment (zero-24) [inline libraries/src/Updater/ConstraintChecker.php]: There are two places to configure stability com_installer and com_joomlaupdate. Has it been verified that this line is only triggered by com_joomlaupdate? Maybe its worth to fork this method into com_joomlaupdate to make it standalone too.

File: administrator/components/com_joomlaupdate/config.xml [hunk 2]
```diff
@@ -19,7 +19,6 @@
 			<!--       All invalid/unsupported/obsolete options equated to default in code with 3.4.0 -->
 			<option value="default">COM_JOOMLAUPDATE_CONFIG_UPDATESOURCE_DEFAULT</option>
 			<option value="next">COM_JOOMLAUPDATE_CONFIG_UPDATESOURCE_NEXT</option>
-			<option value="testing">COM_JOOMLAUPDATE_CONFIG_UPDATESOURCE_TESTING</option>
```
Comment (zero-24) [inline administrator/components/com_joomlaupdate/config.xml:22]: Shoudnt that language string not also be set as deprecated within the language file?

File: libraries/src/Updater/ConstraintChecker.php (hunk 1 above)
Comment (richard67) [reply libraries/src/Updater/ConstraintChecker.php]: https://github.com/bembelimen/joomla-cms/pull/133  should fix this, but I haven't tested it yet together with this PR here. I tested when working on a very similar PR to this here.

Comment (HLeithner) [reply libraries/src/Updater/ConstraintChecker.php]: At this time we don't support extensions with tuf updates but needs to be considered when we are ready for this.

Comment (zero-24) [reply libraries/src/Updater/ConstraintChecker.php]: Sure but isnt the constraintchecker also used for extensions where we support constraints? Or i'm missing Something here?

**Comment by MacJoom** [comment]
I have tested this item :white_check_mark: successfully on ea1b3fb967e28369d03fa1e6d1461144533adb8a<br /><br />From a 5.1 Beta 1 Installation. Without the patch i have the message 'This site can't be updated to Joomla 5.1.0-beta2 .... doesn't meet the minimum requirements...' - with the patch it works!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43147">issues.joomla.org/tracker/joomla-cms/43147</a>.</sub>

**Comment by dautrich** [comment]
I have tested this item :white_check_mark: successfully on ea1b3fb967e28369d03fa1e6d1461144533adb8a<br /><br />Same behavior as described by MacJoom<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43147">issues.joomla.org/tracker/joomla-cms/43147</a>.</sub>

**Comment by richard67** [comment]
RTC as this PR has 2 successful human tests. But it needs more fixes for the updater when using TUF. E.g. the channel ("5.x" or "6.x" depending on Default or Next update source) currently is currently not checked, as far as I can see, and there is one more place where we should use the minimumStability parameter independently from the selected update channel as suggested with my PR for this PR.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43147">issues.joomla.org/tracker/joomla-cms/43147</a>.</sub>

**Comment by MacJoom** [comment]
I have tested both "default" and "next" channel. With the patch it works as expected for both channels. It gives different messages depending on the channel set. "next" channel offers the update to 5.1 beta 2.

File: libraries/src/Updater/ConstraintChecker.php (hunk 1 above)
Comment (bembelimen) [reply libraries/src/Updater/ConstraintChecker.php]: Merged @richard67 PR, thanks for that

File: administrator/components/com_joomlaupdate/config.xml (hunk 2 above)
Comment (bembelimen) [reply administrator/components/com_joomlaupdate/config.xml:22]: Done, thx

**Comment by LadySolveig** [comment]
Thank you @bembelimen and for support to @richard67 and also for testing @MacJoom and @dautrich

**Pull request merged**